framework/src/dgkernels/ADDGKernel.C

```diff
@@ -139,35 +139,74 @@ ADDGKernel::computeElemNeighResidual(Moose::DGResidualType type)
   }
 }
 
+void
+ADDGKernel::computeJacobian()
+{
+  // AD only needs to do one computation for one variable because it does the derivatives all at
+  // once
+  if (!excludeBoundary())
+  {
+    computeElemNeighJacobian(Moose::ElementElement);
+    computeElemNeighJacobian(Moose::NeighborNeighbor);
+  }
+}
+
 void
 ADDGKernel::computeElemNeighJacobian(Moose::DGJacobianType type)
 {
-  const VariableTestValue & test_space =
-      (type == Moose::ElementElement || type == Moose::ElementNeighbor) ? _test : _test_neighbor;
-  const VariableTestValue & loc_phi =
-      (type == Moose::ElementElement || type == Moose::NeighborElement) ? _phi : _phi_neighbor;
+  mooseAssert(type == Moose::ElementElement || type == Moose::NeighborNeighbor,
+              "With AD you should need one call per side");
 
-  if (type == Moose::ElementElement)
-    prepareMatrixTag(_assembly, _var.number(), _var.number());
-  else
-    prepareMatrixTagNeighbor(_assembly, _var.number(), _var.number(), type);
+  const VariableTestValue & test_space = type == Moose::ElementElement ? _test : _test_neighbor;
 
-  auto ad_offset =
-      Moose::adOffset(_var.number(), _sys.getMaxVarNDofsPerElem(), type, _sys.system().n_vars());
+  std::vector<ADReal> residuals(test_space.size(), 0);
 
   for (_qp = 0; _qp < _qrule->n_points(); _qp++)
     for (_i = 0; _i < test_space.size(); _i++)
+      residuals[_i] +=
+          _JxW[_qp] * _coord[_qp] *
```

Comment:
This would save some `Real` math but wouldn't save any `ADReal` math since `computeQpResidual()` has to happen in in the inner loop. Keeping it this way keeps it consistent with what we do elsewhere (except when doing displaced calculations in `ADKernel`)